datafusion/core/tests/dataframe/dataframe_functions.rs

```diff
@@ -186,6 +187,25 @@ async fn test_fn_approx_percentile_cont() -> Result<()> {
 
     assert_batches_eq!(expected, &batches);
 
+    // the arg2 parameter is a complex expr, but it can be evaluated to the literal value
```

Comment:
I think this may be related to aliasing rather than simply complex expressions

```sql
❯ create table t (x int) as values (1), (2);
0 rows in set. Query took 0.001 seconds.

❯ select APPROX_PERCENTILE_CONT(x,  cast(1 as double)) from t;
+--------------------------------------+
| APPROX_PERCENTILE_CONT(t.x,Int64(1)) |
+--------------------------------------+
| 2                                    |
+--------------------------------------+
1 row in set. Query took 0.002 seconds.
```

However, I couldn't express the same alias using SQL. 

I wonder if the expression simplifier code could be generalized to handle this case, so it applied to all arguments, not just the arguments to `APPROX_PERCENTILE_CONT` 🤔

Reply:
yes, the rule of `simplfy_expression` can handle the alisa expr.
But the `cast(1 as float)` will be simplified.
I think this is the diff between the cast and alisa